care/facility/tests/test_prescriptions_api.py

```diff
@@ -113,3 +122,26 @@ def test_create_prn_prescription(self):
             prn_prescription_data,
         )
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_medicine_filter_for_prescription(self):
+        # post 2 prescriptions with different medicines
+        self.client.post(
+            f"/api/v1/consultation/{self.consultation.external_id}/prescriptions/",
+            self.normal_prescription_data,
+        )
+        self.client.post(
+            f"/api/v1/consultation/{self.consultation.external_id}/prescriptions/",
+            self.normal_prescription_data2,
+        )
+
+        # get all prescriptions without medicine filter
+        response = self.client.get(
+            f"/api/v1/consultation/{self.consultation.external_id}/prescriptions/",
+        )
+        self.assertEqual(response.data["count"], 2)
+
+        # get all prescriptions with medicien filter
+        response = self.client.get(
+            f"/api/v1/consultation/{self.consultation.external_id}/prescriptions/?medicine={self.medicine.external_id}",
+        )
+        self.assertEqual(response.data["count"], 1)
```

Comment:
The right test here would be to filter by medicine directly and check if all of the prescriptions returned has that medicine instead of simply checking count.